src/webgldrawer.js

```diff
@@ -491,9 +480,15 @@
         setImageSmoothingEnabled(enabled){
             if( this._imageSmoothingEnabled !== enabled ){
                 this._imageSmoothingEnabled = enabled;
-                this._unloadTextures();
-                this.viewer.world.draw();
+
+                // Todo consider removing old type handlers if _supportedFormats had already types defined,
+                // and remove support for rendering old types...
+                const newFormats = this._setupTextureHandlers();  // re-sets the type to enforce re-initialization
```

Comment:
Maybe support force-refresh instead of this? Seems a bit too complex, but could have some advantages (feels more robust). One thing that happens to come to my mind is that if current processing of invalidation is outdated, this 'force' might not propagate properly.

Reply:
Can you explain why we need extra handling here? My understanding is we just need to re-upload the textures to the GPU; there shouldn't be any data conversion needed.

Reply:
I agree with @iangilman, this is a really confusing way to handle this scenario.

The textures just need to be unloaded and they will automatically be re-loaded which will take account of the new setting. This should not be affected by the cache overhaul at all (and if it is, I think it may be indicative of a design flaw)

Reply:
This on the other hand must be affected by the cachy system: the data needs to be re-uploaded to the gpu. But, in order to do that, you have to go through the data pipeline. You cannot just move data under the nose of the data management system and expect it to behave consistently. Drawer define data convertors, and the system drives the data. As I said, it might be better here to for example add support for data re-validation (~re-upload), but this approach ensures that you can do this re-upload job parts by parts (not all the data at once), by swapping out the type ID.  It can update the data while the renderer is still drawing. Any improvements on the invalidation pipeline will also show improvements here (e.g. the invalidation tile order heuristic I talked about in comments). Truth is, this filres potentially unnecessary code, so we might want to e.g. skip firing the event, just compute the data conversion handling stuff around.

Reply:
I guess I sort of see what you mean, in that you've defined a converter function that specifically creates a gl texture object using an older value for this setting, and the texture itself is cached rather than the underlying data. Is that right? If so, you'd need a new converter for the new setting.

In my opinion, this is an example of a drawer setting that should ideally be entirely orthogonal to the cache system (and it was, previously)

While I love the new capabilities introduced by the new data pipeline, I don't necessarily love that there is so much added complexity and interdependence being introduced. I like when things have clean separation of concerns as much as possible...

In the meantime, is there any reason that re-learning converters during this swap would be a problem? I.e. the code here gets called every time the setting changes. Do these overwrite the old converters or just get added to a list somewhere that keeps growing?
```
            $.convertor.learn("context2d", c2dTexType, (t, d) => tex2DCompatibleLoader(t, d.canvas), 1, 3);
            $.convertor.learn("image", imageTexType, tex2DCompatibleLoader, 1, 3);

            $.convertor.learnDestroy(c2dTexType, tex2DCompatibleDestructor);
            $.convertor.learnDestroy(imageTexType, tex2DCompatibleDestructor);
```

Reply:
> In my opinion, this is an example of a drawer setting that should ideally be entirely orthogonal to the cache system (and it was, previously)

IMHO not possible, if you have three independent parties that can touch the data. One party cannot indenepdently decide to do something under the hood and expect everything still work. That was the problem - it was previously independent. It was simple. And it broke all the data modification plugins (like the filtering one).

Reply:
> In the meantime, is there any reason that re-learning converters during this swap would be a problem?

No, it can be removed. But this is IMHO such a tiny optimization that can be done in subsequent steps, or I am also waiting whether we decide we want to do it differently. And in case we would decide to e.g. do some delay of tile invalidation that has some priority heuristic, we might, depending on the design, actually need to do this. But I guess I would still ensure that the await on invalidation resolves only once all tiles of that particular batch were done.